packages/eslint-plugin/src/rules/explicit-function-return-type.ts

```diff
@@ -59,11 +66,36 @@ export default util.createRule<Options, MessageIds>({
       allowHigherOrderFunctions: true,
       allowDirectConstAssertionInArrowFunctions: true,
       allowConciseArrowFunctionExpressionsStartingWithVoid: false,
+      allowedNames: [],
     },
   ],
   create(context, [options]) {
     const sourceCode = context.getSourceCode();
-
+    function isAllowedName(
+      node:
+        | TSESTree.ArrowFunctionExpression
+        | TSESTree.FunctionExpression
+        | TSESTree.FunctionDeclaration,
+    ): boolean {
+      if (
+        node.type === AST_NODE_TYPES.ArrowFunctionExpression ||
+        node.type === AST_NODE_TYPES.FunctionExpression
+      ) {
+        const parent = node.parent;
+        return (
+          parent?.type === AST_NODE_TYPES.VariableDeclarator &&
+          parent?.id.type === AST_NODE_TYPES.Identifier &&
```

Comment:
Nit: if `parent?.type === AST_NODE_TYPES.VariableDeclarator`, we know `parent` is defined:

```suggestion
          parent.id.type === AST_NODE_TYPES.Identifier &&
```